### ISSUE TITLE
Unexpected buffer in "compilerOutput"

### ISSUE THREAD

**Issue by tustin2121** [furkleindustries/inklecate-node#3]
I was attempting to use this module (jury rigged via a custom task in grunt):
```js
inklecate({ inputFilepath: file.src[0] }).then((output)=>{
	for (let line of output.compilerOutput) {
		if (line.startsWith('ERROR')) {
			grunt.log.error(line);
		} else if (line.startsWith('WARNING')) {
			grunt.log.warn(line);
		} else {
			grunt.log.writeln(line);
		}
	}
	if (!output.text) throw new Error('No story output!');
	const fCont = `var storyContents = ${JSON.stringify(output.storyContent)}`;
	grunt.log.writeln('Writing to file: '+file.dest);
	grunt.file.write(file.dest, fCont);
})
```

However, I kept getting errors when I tried to iterate over compilerOutput. When I printed it out, I discovered this: `[{"type":"Buffer","data":[239,187,191]}]`. Seeing as the typescript interface says that only strings are supposed to be in there, I figured I'd write this up here.

Incidentally, the actual console output seems to have been lost entirely. There's several TODOs in my test story, of which I see none in the compilerOutput array.

(I also got errors prior to this because the npm install didn't properly mark the `inklecate` executable as executable on MacOS. I can write that up as a separate bug if you'd like. All it took for me to fix that was to `chmod +x inklecate`, but I suspect fixing the npm install is more involved.)

**Comment by furkle** [comment]
oh, cool, thanks so much for reporting this. you're definitely right, that buffer should be stringified before output, there's no value to getting a byte buffer, and the types say otherwise.

that `inklecate` error is actually one i keep forgetting to fix, i've run into it a ton of times. only problem is i barely use linux anymore, so i keep forgetting. mostly it should be as simple as adding that shell command to the npm `postInstall` script, but i'm sure there's a package which will ensure it works well for all the UNIX, POSIX, macOS, etc. environments. i'll try to have a new patch version up by tomorrow afternoon :]

**Comment by furkle** [comment]
also, as to the incidental part. that's probably a bug. unfortunately, this is one of the main problems with using inklecate as a string output program, and allowing the package to manage caching and then deleting temporary JSON files. because that's happening over `stdout`, any `TODO` style output will end up breaking any JSON object. to me the sensible thing seems to be to actually fix the problem with `TODO` showing up in the compiler output, but i also know that it's very much a "breaking-ish" change relative to the `inklecate` binaries, and i should probably document it as such. i'm also happy to hear from you as far as better ways we might serve both objectives at once.

**Comment by tustin2121** [comment]
> because that's happening over `stdout`, any `TODO` style output will end up breaking any JSON object. to me the sensible thing seems to be to actually fix the problem with `TODO` showing up in the compiler output, but i also know that it's very much a "breaking-ish" change relative to the `inklecate` binaries, and i should probably document it as such. 

Well, the thing with the stdout output is that everything related to TODOs or WARNINGS or even ERRORs get printed to standard out by the inklecate binaries. By removing that output (assuming I'm understanding you correctly, what you said here was a bit hard to parse), you're also removing any feedback on if a story fails to compile. It's easy enough to say "fix the input story", but without the feedback needed to do so, that's kinda not possible. This is why I was printing out the console output to the grunt output display to begin with, in case there's something I didn't catch when writing in Inky (or more likely a breaking change I made in VSCode, because of course Inky is flawed in its own ways...).

It is unfortunate that inklecate doesn't have a way to pipe the output json anywhere, that is has to go to a file, but at least that means everything coming over the stdout/stderr is meta information and not story data...

**Comment by furkle** [comment]
i am aware that all of that gets printed to `stdout`. if a story fails to compile, the package definitely produces errors and warnings. i use it to make a very large game, and i promise it definitely breaks, and tells me exactly what line was wrong, whenever i screw up a variable :]

it's also very possible my game uses this package differently than you're using it, and there's bugs preventing the normal functioning. the `TODO` part not showing up at all is part of that, that i will need to fix as a bug.

but unfortunately this package -- at least as i designed it up to now -- isn't really here to duplicate `inklecate`. it's to update it into something that actually works as part of a toolchain, and reserves all that logging to be processed or displayed later. what this package does, entirely on purpose, is the opposite -- everything that comes over `stdout` is story data.

i've actually spoken to joe about adding a potential CLI argument for the actual inklecate binaries which would allow one to stream output rather than write files. for now, you are definitely correct in that this package is fairly awkward to try to use for logging author output.

**Comment by furkle** [comment]
i understand this problem better now. you're actually right, if one is specifying an output file, which is to say emitting the JSON in a file write rather than in `stdout`, one does lose all the compiler data. and that's clearly not the best way i could do it. i'm going to put up a new minor version in which, if one is saving the file, they automatically get all the messages in `compilerOutput`. thanks for letting me know about this, it'll be a useful fix. should be up by tomorrow morning, and i'll post the new version ID here.

**Comment by furkle** [comment]
![image](https://user-images.githubusercontent.com/9000099/88868545-74c16e00-d1de-11ea-81a5-05918995517f.png)

how is this? let me know if you'd like it to be any different, it should include all output entirely, not just TODOs.

**Comment by tustin2121** [comment]
So from what you're saying, it sounds like you're using a custom-built version of inklecate which spits out the json via stdout, which is not something the official version does. I was under the assumption you were simply wrapping the official binary.

If that's the case, I'd suggest modifying your custom inklecate to simply output every error, warning, and TODO line over stderr and reserve stdout only for the json output. There's nothing special about stderr that means printing over it indicates an error occurred - that's what the return code is for. Stderr just another pipe to output text through. It gets used all the time for out-of-band feedback when stdout is being used/piped into something else. TODOs are basically just another kind of warning or error message.

**Comment by tustin2121** [comment]
Thank _you_ for making this module.

**Comment by furkle** [comment]
i've updated this here at https://www.npmjs.com/package/inklecate/v/1.8.0. it passes the tests i have, but please feel free to reopen this if you have any similar problems, or if it's not matching your expectations.
